demo/src/features/debug3d/write.rs

```diff
@@ -1,25 +1,101 @@
-use crate::features::debug3d::{Debug3dDrawCall, Debug3dRenderFeature};
-use rafx::api::{RafxResult, RafxVertexBufferBinding};
+rafx::declare_render_feature_write_job!();
+
+use rafx::api::RafxPrimitiveTopology;
+use rafx::api::RafxVertexBufferBinding;
+use rafx::framework::{VertexDataLayout, VertexDataSetLayout};
+
+/// Vertex format for vertices sent to the GPU
+#[derive(Clone, Debug, Copy, Default)]
+#[repr(C)]
+pub struct Debug3DVertex {
```

Comment:
Since this is being used from the prepare job and the write job, I would move it to the base mod.rs - it's doesn't clearly "belong" to either one exclusively

Reply:
I put this into the `write.rs` file because it's part of the the writer's interface. The reason why the `prepare` job uses it is because the `prepare` job is fulfilling that interface.

![image](https://user-images.githubusercontent.com/4888530/114216866-550d4880-991c-11eb-89f1-52fae1d8dec8.png)

Reply:
We have cases where the Vertex is used by an importer (see `TileLayerVertex` and `MeshVertex`). So I still think that it should be a top-level object in the feature in the mod.rs.

Reply:
For the mesh and tile layer, I agree, those should be locally in a clearly public location. The `tile_layer` `TileLayerVertex` is currently in the `public` directory, but pending https://github.com/aclysma/rafx/pull/122#discussion_r610996877 I'll move it into the top-level folder.

I'm less convinced on `Debug3DVertex` because it's not used outside of the feature, and I'm not sure in what scenario it would be.

Reply:
* if it's public I think we agree the vertex declaration can be in the top-level mod.rs (it's very small, doesn't justify its own .rs file)
 * I think we also agree it's best to have it in a consistent place
 * So I personally think we should always put it in the top-level mod.rs so it's in the same place for all features and just add pub as needed

Reply:
Now that I've done a restructure of everything, I kind of like the super minimal `mod.rs` file. It's _just_ the feature declaration, private `use` includes, and the public API reexport.

If the goal is to have it all in a consistent place, I would honestly prefer moving it to the `write.rs` file and just re-exporting it as public in the top-level `mod.rs` when needed (e.g. for `MeshVertex` or `TileLayerVertex`). I think that's nicer because it's next to the vertex declaration too.

Reply:
e.g.

```rust
mod jobs;
use jobs::*;

use rafx::render_feature_mod_prelude::*;
rafx::declare_render_feature!(TileLayerRenderFeature, TILE_LAYER_FEATURE_INDEX);

// Public API

mod plugin;
pub use plugin::*;

mod tile_layer_render_node_set;
pub use tile_layer_render_node_set::*;

mod tile_layer_resource;
pub use tile_layer_resource::*;

pub use jobs::TileLayerVertex;
```